packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -749,6 +793,87 @@ class _AmPmButton extends StatelessWidget {
   }
 }
 
+class _AmPmSemantics extends SingleChildRenderObjectWidget {
+  const _AmPmSemantics({
+    super.child,
+    required this.minSemanticHeight,
+    required this.alignment,
+    required this.checked,
+  });
+
+  final double minSemanticHeight;
+  final Alignment alignment;
+  final bool checked;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderAmPmSemantics(minSemanticHeight, alignment, checked);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderAmPmSemantics renderObject) {
+    renderObject.minSemanticHeight = minSemanticHeight;
+    renderObject.alignment = alignment;
+    renderObject.checked = checked;
+  }
+}
+
+class _RenderAmPmSemantics extends RenderProxyBox {
+  _RenderAmPmSemantics(this._minSemanticHeight, this._alignment, this._checked, [RenderBox? child])
+    : super(child);
+
+  double get minSemanticHeight => _minSemanticHeight;
+  double _minSemanticHeight;
+  set minSemanticHeight(double value) {
+    if (_minSemanticHeight == value) {
+      return;
+    }
+    _minSemanticHeight = value;
+    markNeedsSemanticsUpdate();
+  }
+
+  Alignment get alignment => _alignment;
+  Alignment _alignment;
+  set alignment(Alignment value) {
+    if (_alignment == value) {
+      return;
+    }
+    _alignment = value;
+    markNeedsSemanticsUpdate();
+  }
+
+  bool get checked => _checked;
+  bool _checked;
+  set checked(bool value) {
+    if (_checked == value) {
+      return;
+    }
+    _checked = value;
+    markNeedsSemanticsUpdate();
+  }
+
+  @override
+  void describeSemanticsConfiguration(SemanticsConfiguration config) {
+    super.describeSemanticsConfiguration(config);
+    config.isChecked = checked;
+    config.isInMutuallyExclusiveGroup = true;
+    config.isButton = true;
+  }
+
+  @override
+  Rect get semanticBounds {
```

Comment:
If we increase the SizedBox above the Material widget do we still need this trick?